integrationtest/src/main/java/software/amazon/timestream/integrationtest/DatabaseMetaDataMultiDBMultiTBIntegrationTest.java

```diff
@@ -0,0 +1,301 @@
+   *
+   * @param tablePattern the table pattern to be tested
+   * @param index        index of table name in Constants.MULTI_DB_MUTLI_TB_TABLE_NAMES2
+   * @throws SQLException the exception thrown
+   */
+  @ParameterizedTest
+  @CsvSource(value = {
+    "%-Test%, 0",
+    "%!_02!_01' escape '!, 0",
+    "_ntegration_Test_Ta1ble_0__01, 0",
+    "%_Table_02_02, 1",
+    "%gration.-Te-st%, 1",
+    "%-Te-st!_%' escape '!, 1"
+  })
+  @DisplayName("Test retrieving Integration-Test_Ta1ble_02_01, Integration.-Te-st_Table_02_02 from Integr.ation_Test_Ta_ble_02.")
+  void testTablesWithPatternFromDB2(final String tablePattern, final int index) throws SQLException {
+    try (ResultSet tableResultSet = metaData.getTables(null, Constants.MULTI_DB_MUTLI_TB_DATABASES_NAMES[1], tablePattern, null)) {
+      Assertions.assertTrue(tableResultSet.next());
+      Assertions.assertEquals(Constants.MULTI_DB_MUTLI_TB_TABLE_NAMES2[index], tableResultSet.getObject("TABLE_NAME"));
+    }
+  }
+
+  /**
+   * Test getTables returns tables from JD-BC_Integration.Test_DB_003 when given matching patterns.
+   *
+   * @param tablePattern  the table pattern to be tested
+   * @param schemaPattern the database pattern to be tested
+   * @param index         index of table name in Constants.MULTI_DB_MUTLI_TB_TABLE_NAMES2
+   * @throws SQLException the exception thrown
+   */
+  @ParameterizedTest
+  @CsvSource(value = {
+    "%-BC_Integration-Test_Ta1%, JD-BC%, 0",
+    "%!_03!_01' escape '!, %ion.T%, 0",
+    "JD-BC__ntegration-Test_Ta1ble_0__01, %DB!_003' escape '!, 0",
+    "%.-Te-st_T%, %ion.T%, 1",
+    "%03_02, JD-BC%, 1",
+    "JD-BC!_Integration.-Te-%' escape '!, %DB!_003' escape '!, 1",
+    "%--Test%, %ion.T%, 2",
+    "%3_03, JD-BC%, 2",
+    "%t2!_T%' escape '!, %DB!_003' escape '!, 2",
+    "%0-Te-st%, JD-BC%, 3",
+    "%_3_04, %ion.T%, 3",
+    "%a.ble%' escape '!, %DB!_003' escape '!, 3"
+  })
+  @DisplayName("Test retrieving JD-BC_Integration-Test_Ta1ble_03_01, JD-BC_Integration.-Te-st_Table_03_02, JD-BC_Integration--Test2_Table_03_03, JD-BC_Integration0-Te-st_Ta.ble_03_04 from JD-BC_Integration.Test_DB_003.")
+  void testTablesWithPatternFromDB3WithPattern(final String tablePattern, final String schemaPattern, final int index) throws SQLException {
+    try (ResultSet tableResultSet = metaData.getTables(null, schemaPattern, tablePattern, null)) {
+      Assertions.assertTrue(tableResultSet.next());
+      Assertions.assertEquals(Constants.MULTI_DB_MUTLI_TB_TABLE_NAMES3[index], tableResultSet.getObject("TABLE_NAME"));
+    }
+  }
+
+  /**
+   * Test getTables returns tables from JD-BC_Integration.Test_DB_003 when given matching patterns.
+   *
+   * @param tablePattern the table pattern to be tested
+   * @param index        index of table name in Constants.MULTI_DB_MUTLI_TB_TABLE_NAMES2
+   * @throws SQLException the exception thrown
+   */
+  @ParameterizedTest
+  @CsvSource(value = {
+    "%-BC_Integration-Test_Ta1%, 0",
+    "%!_03!_01' escape '!, 0",
```

Comment:
I think the test purpose of this one is same as testDB1 and testDB2. We only have %, _ as wildcard. But so many tests seem redunant. Also they are a little hard to read and maintain.

Reply:
What I had in mind was to test all tables from the multiple databases. I feel if we do not test retrieving all tables then we're losing the point of creating those tables. I believe it is best to maintain testing all tables that we created, but I will remove some test cases of patterns: e.g., test 1 pattern combination for one index instead of testing 3 pattern combinations.

Reply:
I agree to test all tables but for pattern related test it could be done in one test instead of testing it in different case. More tests may not be a good thing. Test efficiency and test script maintainbility is also a thing that needs to be considered. In my opnion we have too many redunant tests for patterns.

Reply:
I had removed some tests, please let me know if you have more comments

Reply:
I combined all pattern tests to two. One for testing with table patterns only, another for testing with both table patterns and schema patterns